src/Parser/Core/Modules/Items/BFA/Dungeons/BalefireBranch.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import { formatDuration } from 'common/format';
+import { calculatePrimaryStat } from 'common/stats';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Abilities from 'Parser/Core/Modules/Abilities';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+const ACTIVATION_COOLDOWN = 90; // seconds
+
+const STACKS_START = 100;
+const DURATION_TO_DECAY = 20; // seconds
+const STACKS_LOST_PER_SEC = 5;
```

Comment:
nit: Technically more correct would be `STACKS_START / DURATION_TO_DECAY` since that most likely matches the way it's implemented in-game